sklearn/semi_supervised/label_propagation.py

```diff
@@ -173,10 +173,8 @@ class labels
         """
         check_is_fitted(self, 'X_')
 
-        if sparse.isspmatrix(X):
-            X_2d = X
-        else:
-            X_2d = np.atleast_2d(X)
+        X_2d = check_array(X, accept_sparse = ['csc', 'csr', 'coo', 'dok',
```

Comment:
does it actually work with a dok matrix? I'd be surprised but a common test should error if not... hum... can you please try it?